### PR TITLE
Add MMADetectorTimeIntervals

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds concept of MMADetectorTimeIntervals. Useful for tracking the on/off state for multi-messenger detectors.

### CODE REVIEW

**Review by guynir42 — COMMENTED** [review]
Code looks fine. 
I'm not sure I understand what a "segment" is in this context. In general you should specify more details in the PR comment. Beyond that, I'm not sure it will be obvious to users what a segment is. I went through the code and still don't understand what it does or why it is called a segment (I would imaging a segment to be maybe one of two or three GW detector or one of the arms of an interferometer, so I was very surprised to see it is a segment in time).

**Comment by mcoughlin** [comment]
@guynir42 Good point, I added some more explanation in the model.

**Comment by guynir42** [comment]
@mcoughlin I believe we can find a better name than "segment" if we really put our mind to it.

**Comment by mcoughlin** [comment]
@guynir42 TimeSegments?

**Comment by guynir42** [comment]
@mcoughlin how about TimeInterval?

File: skyportal/handlers/api/mmadetector.py [hunk 1]
```diff
@@ -610,3 +613,396 @@ def delete(self, spectrum_id):
...
+                if {g.id for g in groups} != set(group_ids):
+                    return self.error(
+                        f'Cannot find one or more groups with IDs: {group_ids}.'
+                    )
+
+                if groups:
+                    time_interval.groups = time_interval.groups + groups
+
+            if 'time_interval' in data:
+                setattr(time_interval, 'time_interval', data['time_interval'])
```
Comment (guynir42) [inline skyportal/handlers/api/mmadetector.py:944]: Why not `time_interval.time_interval = data['time_interval']` ?

File: skyportal/models/group.py [hunk 2]
```diff
@@ -365,6 +365,15 @@ class Group(Base):
...
     )
 
+    mmadetector_time_intervals = relationship(
+        "MMADetectorTimeInterval",
+        secondary="group_mmadetector_time_intervals",
+        back_populates="groups",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+        doc='The MMADetector time intervals visible to this group.',
+    )
```
Comment (guynir42) [inline skyportal/models/group.py:375]: Just wondering, do we really need to associate a time interval with a group? Is there expected to be permissioning on time intervals?

**Review by guynir42 — APPROVED** [review]
Looks good. Couple of small comments/questions:

File: skyportal/models/group.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/models/group.py:375]: I am kind of envisioning that one might want to post particular data subsets (i.e. detections or something), that might have restrictive access.

File: skyportal/handlers/api/mmadetector.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/mmadetector.py]: Good idea.

**Pull request merged**